eZ/Bundle/EzPublishCoreBundle/Routing/DefaultRouter.php

```diff
@@ -95,7 +96,13 @@ public function generate($name, $parameters = array(), $referenceType = self::AB
             unset($parameters['siteaccess']);
         }
 
-        $url = parent::generate($name, $parameters, $referenceType);
+        try {
+            $url = parent::generate($name, $parameters, $referenceType);
+        } catch (RouteNotFoundException $e) {
+            // Switch back to original context, for next links generation.
+            $this->setContext($originalContext);
+            throw $e;
+        }
```

Comment:
But then it would have to indent all the code as opposed to return early approach